comp/core/tagger/tagger.go

```diff
@@ -360,8 +362,14 @@ func (t *TaggerClient) ResetCaptureTagger() {
 func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
 	cardinality := taggerCardinality(originInfo.Cardinality)
 
+	// If dogstatsdUnifiedEnabled is disabled, we use DogStatsD's Legacy Origin Detection.
+	// TODO: remove this when dogstatsd_origin_detection_unified is enabled by default
+	if !t.datadogConfig.dogstatsdUnifiedEnabled {
+		originInfo.ProductOrigin = taggertypes.ProductOriginDogStatsDLegacy
```

Comment:
Instead of overriding the config, I think the caller should provide the right `ProductOrigin`.